### PR TITLE
Documentation for SonarCloud configuration with CI

### PR DESCRIPTION
**Pull request by tappasarn**

### CODE REVIEW

File: documentation/codeQuality/README.md [hunk 1]
```diff
@@ -1,2 +1,233 @@
...
+
+        CurrencyFormatter currencyFormatter = new CurrencyFormatter();
+
+        System.out.println(currencyFormatter.format(amount));
+    }
+```
+12. Rerun the analyze command again !
+
+## More problems !
+We have said early on that SonarCloud does not only analyze your code base, however it actually reports the unit test coverage of our project.
```
Comment (hibikiledo) [inline documentation/codeQuality/README.md]: earlier that

,but also report unit test coverage

File: documentation/codeQuality/README.md [hunk 2]
```diff
@@ -1,2 +1,233 @@
...
+
+        System.out.println(currencyFormatter.format(amount));
+    }
+```
+12. Rerun the analyze command again !
+
+## More problems !
+We have said early on that SonarCloud does not only analyze your code base, however it actually reports the unit test coverage of our project. 
+
+Having unit test coverage helps us to have visibility on how trustful our software is. With test coverage report, we should be able to easily tell where unit test are missing and priority them in the development team.
```
Comment (hibikiledo) [inline documentation/codeQuality/README.md]: helps us to have visibility on -> shows us

and priority them -> and prioritise them

File: documentation/codeQuality/README.md [hunk 3]
```diff
@@ -1,2 +1,233 @@
...
+    id "jacoco"
+}
+```
+
+2. To see the test report, run the following command
+```
+gradle test jacocoTestReport
+```
+
+3. By default, you test report will be located under .build/reports/jacoco
```
Comment (hibikiledo) [inline documentation/codeQuality/README.md]: your test report

File: documentation/codeQuality/README.md [hunk 4]
```diff
@@ -1,2 +1,233 @@
...
+    }
+}
+```
+
+## What have we done so far ?
+Up until now, we are able to analyze our code base using the service provide by SonarCloud. However, the analyze process is not yet automated. We are back to the same issue that we used to have with unit tests early in the beginning of this workshop. Every time we have made some changes into our code, we will have to use SonarCloud's analyze command and manually run it by ourselves. I'm sure you would not want to do it. Most of us might not even want to remember the command itself !
+
+## Integrate SonarCould with GitHub and Continuous Integration
+
+The objective for this section is to block the pull request that does not pass SonarCloud's standard from merging into the master branch. So the owners of the pull requests know that their PR need to be taken care of.
```
Comment (hibikiledo) [inline documentation/codeQuality/README.md]: Maybe, we should write pull requests(PR) since we use PR later in the sentence.

Comment (hibikiledo) [reply documentation/codeQuality/README.md]: PRs 

^ is this valid lol

File: documentation/codeQuality/README.md [hunk 5]
```diff
@@ -1,2 +1,233 @@
...
+
+cache:
+  directories:
+    - '$HOME/.m2/repository'
+    - '$HOME/.sonar/cache'
+    - '$HOME/.gradle'
+    - '.gradle'
+
+```
+5. At this point, if you are still enable travis ci build process for branches, you can push your project into GitHub sever and see SonarCloud is being execute by TravisCI.
```
Comment (hibikiledo) [inline documentation/codeQuality/README.md]: if you are still enable travis ci build process for branches

^
Did we ask them to disable it at all ?

Comment (hibikiledo) [reply documentation/codeQuality/README.md]: sever -> server

File: documentation/codeQuality/README.md [hunk 6]
```diff
@@ -1,2 +1,233 @@
...
+    
+    int target = -5;
+    int num = 3;
+
+    target =+ num; // Noncompliant; target = 3
+
+    System.out.println(currencyFormatter.format(amount));
+}
+```
+9. Make a pull request against the master branch.
```
Comment (hibikiledo) [inline documentation/codeQuality/README.md]: Should we add a picture for this ?

File: documentation/codeQuality/README.md [hunk 7]
```diff
@@ -1,2 +1,233 @@
...
+```
+
+2. To see the test report, run the following command
+```
+gradle test jacocoTestReport
+```
+
+3. By default, you test report will be located under .build/reports/jacoco
+
+4. Now let's return the SonarCloud analyze command. Your coverage result should be present on the SonarCloud's site.
```
Comment (hibikiledo) [inline documentation/codeQuality/README.md]: return -> rerun

File: documentation/codeQuality/README.md [hunk 8]
```diff
@@ -1,2 +1,233 @@
...
+
+5. In Addition, you can fix the version of JaCoCo by adding this section into build.gradle
```
Comment (hibikiledo) [inline documentation/codeQuality/README.md]: may be change 

fix -> specify

When read this sentence for the first time, I thought I am going to fix something.

File: documentation/codeQuality/README.md [hunk 9]
```diff
@@ -1,2 +1,233 @@
...
+    3. Add your SonarCloud's token in Environment variable section. For simplicity of the workshop, let's give the token the same name, SONAR_TOKEN.
+    ![travis add token](https://user-images.githubusercontent.com/11821799/46256317-361fd700-c4d3-11e8-99da-303fcdfefb82.png)
+
+2. Get SonarCloud.io organization key
+    1. Visit this https://sonarcloud.io/account/organizations
+    
+    2. You should see the organization key near by your project.
+    ![sonar org key](https://user-images.githubusercontent.com/11821799/46256416-da564d80-c4d4-11e8-8c82-b8076133ce55.png)
+
+3. Create sonar-project.properties
```
Comment (hibikiledo) [inline documentation/codeQuality/README.md]: Just create an empty file ?

File: documentation/codeQuality/README.md (hunk 1 above)
Comment (tappasarn) [reply documentation/codeQuality/README.md]: fixed

File: documentation/codeQuality/README.md (hunk 2 above)
Comment (tappasarn) [reply documentation/codeQuality/README.md]: fixed

File: documentation/codeQuality/README.md (hunk 3 above)
Comment (tappasarn) [reply documentation/codeQuality/README.md]: fixed

File: documentation/codeQuality/README.md (hunk 4 above)
Comment (tappasarn) [reply documentation/codeQuality/README.md]: done

File: documentation/codeQuality/README.md (hunk 5 above)
Comment (tappasarn) [reply documentation/codeQuality/README.md]: we showed him how to disable it in the earlier topic about CI

Comment (tappasarn) [reply documentation/codeQuality/README.md]: fixed typo :)

File: documentation/codeQuality/README.md (hunk 7 above)
Comment (tappasarn) [reply documentation/codeQuality/README.md]: done

File: documentation/codeQuality/README.md (hunk 9 above)
Comment (tappasarn) [reply documentation/codeQuality/README.md]: my mistake, we dont really need this file. deleted this line 👍

File: documentation/codeQuality/README.md (hunk 8 above)
Comment (tappasarn) [reply documentation/codeQuality/README.md]: fixed

**Pull request merged**